src/Data/List/Linear.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE LinearTypes #-}
```

Comment:
Data.List can get long. Could you break this up into sections like in the base counterpart. Maybe look at the format of using `-- # Title` and a line of 90 `-------` characters like in [HashMap.hs]( https://github.com/tweag/linear-base/blob/master/src/Data/HashMap/Linear.hs).